core/IQueue.go

```diff
@@ -0,0 +1,111 @@
+
+	// Add adds the given item to the end of the queue. Operation is successful only
+	// if queue has required capacity.
+	// Add returns true if the item is added, false otherwise.
+	Add(item interface{}) (added bool, err error)
+
+	// AddAll adds all the items. If items slice changes during this operation, behaviour is unspecified.
+	// AddAll returns true if the queue has changed, false otherwise.
+	AddAll(items []interface{}) (changed bool, err error)
+
+	// AddItemListener adds an item listener for this queue.
+	// The listener will be invoked for any add/remove item event.
+	// AddItemListener returns the registrationID of the listener.
+	AddItemListener(listener interface{}, includeValue bool) (registrationID *string, err error)
+
+	// Clear removes all of the items in this queue.
+	Clear() (err error)
+
+	// Contains returns true if the given item is in this queue.
+	Contains(item interface{}) (found bool, err error)
+
+	// ContainsAll returns true if this queue contains all the given items, false otherwise.
+	ContainsAll(items []interface{}) (foundAll bool, err error)
+
+	// DrainTo removes all items in this queue and add them to the end of given slice.
+	// DrainTo returns the number of moved items.
+	DrainTo(slice []interface{}, maxElements int32) (movedAmount int32, err error)
+
+	// IsEmpty returns true if this queue is empty, false otherwise.
+	IsEmpty() (empty bool, err error)
+
+	// Offer inserts the given item to the end of the queue if there is room.
+	// If queue is full, offer operation fails.
+	// Offer returns true if the item is added, false otherwise.
+	Offer(item interface{}) (added bool, err error)
+
+	// OfferWithTimeout inserts the given item to the end of the queue if there is room, otherwise
+	// it waits for timeoutInMilliSeconds milliseconds before returning false.
+	// OfferWithTimeout returns true if the item is added, false otherwise.
+	OfferWithTimeout(item interface{}, timeoutInMilliSeconds int64) (added bool, err error)
+
+	// Peek retrieves, but does not remove the head of this queue.
+	// Peek returns the head of this queue, nil if the queue is empty.
+	Peek() (item interface{}, err error)
+
+	// Poll retrieves and removes the head of this queue.
+	// Poll returns the head of this queue, nil if the queue is empty.
+	Poll() (item interface{}, err error)
+
+	// PollWithTimeout retrieves and removes the head of this queue,
+	// if the queue is empty it waits timeoutInMilliSeconds milliseconds before
+	// returning nil.
+	// PollWithTimeout returns the head of this queue, nil if the queue is empty after timeoutInMilliSeconds milliseconds.
+	PollWithTimeout(timeoutInMilliSeconds int64) (item interface{}, err error)
+
+	// Put inserts the item at the end of this queue. It blocks until
+	// space becomes available if necessary.
+	Put(item interface{}) (err error)
+
+	// RemainingCapacity returns the number of additional items this queue can contain.
+	RemainingCapacity() (remainingCapacity int64, err error)
```

Comment:
this returns int32 on the Java side. see https://github.com/hazelcast/hazelcast/blob/master/hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientQueueProxy.java#L208